src/rdiff_backup/SetConnections.py

```diff
@@ -107,43 +108,34 @@ def parse_file_desc(file_desc):
 
     """
 
-    def check_len(i):
-        if i >= len(file_desc):
-            raise SetConnectionsException(
-                "Unexpected end to file description %s" % file_desc)
-
-    host_info_list, i, last_was_quoted = [], 0, None
-    file_desc = os.fsencode(
-        file_desc)  # paths and similar must always be bytes
-    while 1:
-        if i == len(file_desc):
-            # make sure paths under Windows use / instead of \
-            if os.path.altsep:  # only Windows has an alternative separator for paths
-                file_desc = file_desc.replace(os.fsencode(os.path.sep), b'/')
-            return (None, file_desc)
-
-        if file_desc[i] == ord(
-                '\\'):  # byte[n] is the numerical value hence ord
-            i = i + 1
-            check_len(i)
-            last_was_quoted = 1
-        elif (file_desc[i] == ord(":") and i > 0
-              and file_desc[i - 1] == ord(":") and not last_was_quoted):
-            host_info_list.pop()  # Remove last colon from name
-            break
-        else:
-            last_was_quoted = None
-        host_info_list.append(file_desc[i:i + 1])
-        i = i + 1
-
-    check_len(i + 1)
+    # paths and similar objects must always be bytes
+    file_desc = os.fsencode(file_desc)
+    # match double colon not preceded by an odd number of backslashes
+    file_match = re.fullmatch(rb"^(?P<host>.*[^\\](?:\\\\)*)::(?P<path>.*)$",
```

Comment:
For a first time, I'm not following you on this. Regex is not well suited to do this kind of parsing. It add complexity to the code without resolving the actual problem.

Reply:
It resolved the actual problem because trailing backslashes are now possible under Windows. It was never the purpose of this function to remove trailing backslashes, it happens later in the code (rpath.normalize), but the previous code in the current function was choking on such trailing backslashes, which it isn't any more.

Regarding usage of regex:
1. did you have a look at the previous code and did you understand it (I didn't really, that's why I rewrote it)? I think that regexp makes the code much clearer than it used to be.
2. even the python documentation says it: https://docs.python.org/3/library/re.html#simulating-scanf - pattern matching is the way to go, unless you want to write your own parser (which is what the previous code did).
3. the code/regex could be _a lot_ cleaner/simpler but the man page in the REMOTE OPERATION section, in the 3rd paragraph, starting with "Concerning quoting", states quite a lot of complex quoting rules, which makes the regex more complex than it could be.

OK, let me ask on the mailing list if someone is using the extended quoting capabilities, then the code could become much simpler.

Reply:
As per the discussion on the mailing list, we need the current complexity to potentially support IPv6 and its colons. I'm happy to address them if you have further concerns, but I need more concrete questions and feedback.

Reply:
Let keep the regex then.